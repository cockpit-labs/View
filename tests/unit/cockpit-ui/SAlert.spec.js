import { shallowMount } from '@vue/test-utils'
import SAlert from '@/plugins/cockpit-ui/components/alert/SAlert'

describe('Alert', () => {
  const wrapper = shallowMount(SAlert, {
    slots: {
      title: 'Error 500',
      content: 'Cum haec taliaque sollicitas eius aures everberarent expositas semper eius modi rumoribu'
    },
    mocks: {
      $t: key => key
    }
  })

  it('should not be opened by default', () => {
    const alert = wrapper.find('.s-alert')
    expect(alert.exists()).toBe(false)
  })

  it('should have "type" prop equals to "alert"', () => {
    expect(wrapper.props('type')).toBe('alert')
  })

  it('can be open', async () => {
    // wrapper.vm.open()
    wrapper.setData({
      opened: true,
      dialogVisible: true
    })

    await wrapper.vm.$nextTick()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('can be close', async () => {
    wrapper.setData({
      opened: false
    })
    await wrapper.vm.$nextTick()
    const alert = wrapper.find('.s-alert')
    expect(alert.exists()).toBe(false)
    // const alert = wrapper.find('.s-alert')
    // expect(alert.exists()).toBe(false)
  })
  // it('should emit click event on click action button', async () => {
  //   const button = wrapper.find('sbutton-stub')
  //   const spy = jest.fn()
  //   wrapper.vm.$on('click', spy)
  //   button.vm.$emit('click')
  //   await wrapper.vm.$nextTick()
  //   expect(spy).toHaveBeenCalledTimes(1)
  // })
})
