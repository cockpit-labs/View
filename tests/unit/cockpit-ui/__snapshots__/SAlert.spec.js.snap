// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert can be open 1`] = `
<transition-stub
  name="overlay"
>
  <div
    class="s-alert overlay alert"
  >
    <transition-stub
      name="dialog"
    >
      <div
        class="dialog"
      >
        <div
          class="picto"
        >
          <sicon-stub
            name="bell"
          />
        </div>
         
        <div
          class="header"
        >
          Error 500
        </div>
         
        <div
          class="body"
        >
          Cum haec taliaque sollicitas eius aures everberarent expositas semper eius modi rumoribu
        </div>
         
        <div
          class="footer"
        >
          <sbutton-stub
            iconposition="right"
          >
            ok
          </sbutton-stub>
        </div>
      </div>
    </transition-stub>
  </div>
</transition-stub>
`;
